extensions/src/main/java/org/pentaho/platform/web/http/api/resources/FileResource.java

```diff
@@ -938,9 +938,39 @@ public Response doSetContentCreator( @PathParam ( "pathId" ) String pathId, Repo
       @ResponseCode ( code = 500, condition = "Unable to update locale properties due to some other error." ) } )
   public Response doSetLocaleProperties( @PathParam ( "pathId" ) String pathId, @QueryParam ( "locale" ) String locale,
                                          List<StringKeyStringValueDto> properties ) {
+    boolean success = true;
+    String responseMsg = "";
     try {
+      /*
+       [BI-14087] If one of the properties is to change the file/folder name, it is also necessary to run this command to
+       get the file's path to update as well. We need to remove it from the properties, because it can affect future
+       renames. We need to keep it consistent.
+      */
+      String[] pathSplit = pathId.split( ":" );
+      boolean isFileTitleFound = false;
+      StringKeyStringValueDto fileTitle = null;
+      for ( StringKeyStringValueDto property : properties ) {
+        // Check to see if the file.title is listed here, it may be used to rename the file/folder
+        if ( property.getKey().equals( "file.title" ) ) {
+          // Check to see if the file title is different than the current file/folder, if it is, we need to move it.
+          responseMsg = "Unable to rename File/Folder";
```

Comment:
This response message should be in a properties file so it can be translated.

Reply:
Added message to messages.properties